plugins/usertiming.js

```diff
@@ -0,0 +1,110 @@
+/**
+\file usertiming.js
+Plugin to collect metrics from the W3C User Timing API.
+For more information about User Timing,
+see: http://www.w3.org/TR/user-timing/
+
+This plugin is dependent on the UserTimingCompression library
+see: https://github.com/nicjansma/usertiming-compression.js
+UserTimingCompression must be loaded before this plugin's init is called.
+*/
+
+(function() {
+
+	BOOMR = BOOMR || {};
+	BOOMR.plugins = BOOMR.plugins || {};
+
+	if (BOOMR.plugins.UserTiming) {
+		return;
+	}
+
+	/**
+	 * @returns String compressed user timing data that occurred since the last call
+	 */
+	function getUserTiming() {
+		var map, res, now = BOOMR.now();
+
+		map = UserTimingCompression.getCompressedUserTiming(impl.options);
+		res = UserTimingCompression.compressForUri(map);
+		impl.options["from"] = now;
+
+		return res;
+	}
+
+	var impl = {
+		complete: false,
+		initialized: false,
+		supported: false,
+		options: {"from": 0},
+
+		addEntriesToBeacon: function() {
+			var r;
+
+			if (this.complete) {
+				return;
+			}
+
+			BOOMR.removeVar("usertiming");
+			r = getUserTiming();
+			if (r) {
+				BOOMR.info("User Timing API entries found since the last beacon", "usertiming");
+				BOOMR.addVar({
+					"usertiming": r
+				});
+			}
+
+			this.complete = true;
+			BOOMR.sendBeacon();
```

Comment:
Since this is called on `before_beacon`, we should not call `BOOMR.sendBeacon()` here.